src/Conversion/ONNXToTorch/NN/GemmOp.cpp

```diff
@@ -115,39 +116,71 @@ struct ONNXGemmOpToTorchLowering : public ConversionPattern {
     auto C = gemmOp.C();
 
     auto aTensor = getTorchTensor(A, rewriter, context, loc);
-    auto bTensor = getTorchTensor(B, rewriter, context, loc);
-    auto cTensor = getTorchTensor(C, rewriter, context, loc);
-
     auto aType = toTorchType(context, A.getType());
+    auto bTensor = getTorchTensor(B, rewriter, context, loc);
     auto bType = toTorchType(context, B.getType());
-    auto cType = toTorchType(context, C.getType());
     auto resultType = toTorchType(context, gemmOp.getResult().getType());
+    auto cTensor = getTorchTensor(C, rewriter, context, loc);
+
+    // `gemmOp.C()` does not consider batch sizes. Construct a value tensor
+    // from the result type instead. XTen later on expects correct shapes
+    // and does not allow broadcasting.
+    if (mlir::isa<ONNXConstantOp>(C.getDefiningOp()) &&
```

Comment:
Why do you do something special for constants but not other inputs?

Reply:
This fixes the issue for our current use case. Otherwise, we would likely need to fix broadcasting in XTen.